### ISSUE TITLE
New-BcContainer BC20 w1 fails when TestToolkit is included

### ISSUE THREAD

**Issue by JoelJame5** [microsoft/navcontainerhelper#2298]
New-BcContainer command fails with the below error when called with the following switches against insider BC20 W1 image 
-includeTestToolkit -includeTestLibrariesOnly

Seems to be an issue with the W1 image as calling with country set to GB works fine. To get around the issue for the time being we are building the container first without the TestToolkit then adding it afterwards:
 _Restart-BcContainer -containerName $ContainerName
 Start-Sleep -Seconds 5
 Import-TestToolkitToBcContainer -containerName $ContainerName -includeTestLibrariesOnly_  

_```

> Installing Permissions Mock on default
> WARNING: This license is not compatible with this version of Business Central.
> WARNING: UnhandledErrorMessage
> The socket connection was aborted. This could be caused by an error processing your message or a receive timeout being exceeded by the remote host, or a
> n underlying network resource issue. Local socket timeout was '10675199.02:48:05.4775807'.
> at <ScriptBlock>, <No file>: line 18
> at Invoke-ScriptInBcContainer, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\3.0.1\ContainerHandling\Invoke-ScriptInNavContainer.ps1: lin
> e 43
> at Install-BcContainerApp, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\3.0.1\AppHandling\Install-NavContainerApp.ps1: line 39
> at <ScriptBlock>, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\3.0.1\ObjectHandling\Import-TestToolkitToNavContainer.ps1: line 209
> at Import-TestToolkitToBcContainer, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\3.0.1\ObjectHandling\Import-TestToolkitToNavContainer.p
> s1: line 180
> at New-BcContainer, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\3.0.1\ContainerHandling\New-NavContainer.ps1: line 1972
> at <ScriptBlock>, <No file>: line 1
> Import-TestToolkitToBcContainer Telemetry Correlation Id: 15c18064-32ce-414a-bd7c-9969deeb3a31
> New-BcContainer Telemetry Correlation Id: cc37228c-7032-427a-af4f-0633a982e9b9
> The socket connection was aborted. This could be caused by an error processing your message or a receive timeout being exceeded by the remote host, or 
> an underlying network resource issue. Local socket timeout was '10675199.02:48:05.4775807'.
> At C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\3.0.1\ContainerHandling\Invoke-ScriptInNavContainer.ps1:44 char:13
> +             throw $_.Exception.Message
> +             ~~~~~~~~~~~~~~~~~~~~~~~~~~
>     + CategoryInfo          : OperationStopped: (The socket conn...48:05.4775807'.:String) [], RuntimeException
>     + FullyQualifiedErrorId : The socket connection was aborted. This could be caused by an error processing your message or a receive timeout being e 
>    xceeded by the remote host, or an underlying network resource issue. Local socket timeout was '10675199.02:48:05.4775807'

```_

**Comment by freddydk** [comment]
I can confirm that this is a problem in the w1 version and the currently only way to avoid this problem is by restarting the service tier or container between creating the container and importing the test toolkit apps.
The problem is in the Service Tier and I will get somebody in the team to look at that.

**Comment by freddydk** [comment]
Please retry this with insider build 20.0.36207.0 or later, thanks

**Comment by fridrichovsky** [comment]
Hello I tested container crateion
I do NST restart after between New-Container and Import-TestToolkit.
Tests ends with these results. 
```
Test conatiner removed and crated again
```

Parameters:
```
Version: 20.0.36210.0
Country: w1
```

Part of container log:
```
Version: 20.0.36210.0
Platform: 20.0.36169.0
Generic Tag: 1.0.2.1
Container OS Version: 10.0.17763.2237 (ltsc2019)
Host OS Version: 10.0.17763.2213 (ltsc2019)
WARNING: Host OS and Base Image Container OS doesn't match and process isolation is specified. If you encounter issues, you could try to specify -isolation hyperv
Using process isolation
Using locale en-US
```

Error during import test toolkit in first test:
```
Publishing C:\ProgramData\BcContainerHelper\Extensions\NSSProjects-NEXT\12a2aa3f-6ffb-435f-ba3f-449bbcf95493\Microsoft_Tests-Data Exchange.app
WARNING: This license is not compatible with this version of Business Central.
Synchronizing Tests-Data Exchange on tenant default
Installing Tests-Data Exchange on tenant default
WARNING: This license is not compatible with this version of Business Central.
App Microsoft_Tests-Data Exchange.app successfully published
WARNING: This license is not compatible with this version of Business Central.
Publishing C:\ProgramData\BcContainerHelper\Extensions\NSSProjects-NEXT\ce3b1807-44ad-414c-9214-d8cef7e4cd8c\Microsoft_Tests-Dimension.app
WARNING: This license is not compatible with this version of Business Central.
Synchronizing Tests-Dimension on tenant default
Installing Tests-Dimension on tenant default
WARNING: This license is not compatible with this version of Business Central.
App Microsoft_Tests-Dimension.app successfully published
WARNING: This license is not compatible with this version of Business Central.
Publishing C:\ProgramData\BcContainerHelper\Extensions\NSSProjects-NEXT\3a621710-ba88-4100-b7ae-d051f124ae68\Microsoft_Tests-ERM.app
WARNING: This license is not compatible with this version of Business Central.
The background process reported an error with the following message: "The named pipe target process has ended.".
at Invoke-ScriptInBcContainer, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\3.0.2\ContainerHandling\Invoke-ScriptInNavContainer.ps1: line 38
at <ScriptBlock>, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\3.0.2\AppHandling\Publish-NavContainerApp.ps1: line 268
at Publish-BcContainerApp, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\3.0.2\AppHandling\Publish-NavContainerApp.ps1: line 128
at <ScriptBlock>, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\3.0.2\ObjectHandling\Import-TestToolkitToNavContainer.ps1: line 218
at Import-TestToolkitToBcContainer, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\3.0.2\ObjectHandling\Import-TestToolkitToNavContainer.ps1: line 180
at <ScriptBlock>, C:\BCContainers\New-BCContainers.ps1: line 2046
```

Error during import test toolkit in second test:
```
Publishing C:\ProgramData\BcContainerHelper\Extensions\NSSProjects-NEXT\77a31d99-8a30-4b67-8829-76f51572530c\Microsoft_Tests-Dimension.app
WARNING: This license is not compatible with this version of Business Central.
Synchronizing Tests-Dimension on tenant default
Installing Tests-Dimension on tenant default
WARNING: This license is not compatible with this version of Business Central.
App Microsoft_Tests-Dimension.app successfully published
WARNING: This license is not compatible with this version of Business Central.
Publishing C:\ProgramData\BcContainerHelper\Extensions\NSSProjects-NEXT\2292bd89-8314-4b2c-9bae-3d990c4883c2\Microsoft_Tests-ERM.app
WARNING: This license is not compatible with this version of Business Central.
WARNING: UnhandledErrorMessage
The socket connection was aborted. This could be caused by an error processing your message or a receive timeout being exceeded by the remote host, or an underlying network resource issue. Local socket timeout was '10675
199.02:48:05.4775807'.
at <ScriptBlock>, <No file>: line 40
at Invoke-ScriptInBcContainer, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\3.0.2\ContainerHandling\Invoke-ScriptInNavContainer.ps1: line 43
at <ScriptBlock>, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\3.0.2\AppHandling\Publish-NavContainerApp.ps1: line 268
at Publish-BcContainerApp, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\3.0.2\AppHandling\Publish-NavContainerApp.ps1: line 128
at <ScriptBlock>, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\3.0.2\ObjectHandling\Import-TestToolkitToNavContainer.ps1: line 218
at Import-TestToolkitToBcContainer, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\3.0.2\ObjectHandling\Import-TestToolkitToNavContainer.ps1: line 180
at <ScriptBlock>, C:\BCContainers\New-BCContainers.ps1: line 2046
```

**Comment by fridrichovsky** [comment]
I tested same action on different machine and I get socket problem again.

**Comment by fridrichovsky** [comment]
Last information is that I tested container 
Version: 19.3.34541.35002
Country: cz
Type: OnPrem

and I get same error connected with socket timeout.

**Comment by freddydk** [comment]
@fridrichovsky that is a different problem, not related to the problem in this issue.
Please try with hyperv (as suggested in the output) and if it still fails, create a new issue with full repro script and full output.

**Comment by fridrichovsky** [comment]
@freddydk  I am not sure that it is different problem, but I create new issue. Thank you.

**Comment by freddydk** [comment]
This issue is aboutthe latest insider would give you this problem if you didn't restart the container before importing test toolkit.

```
Installing Permissions Mock on default
WARNING: This license is not compatible with this version of Business Central.
WARNING: UnhandledErrorMessage
The socket connection was aborted. This could be caused by an error processing your message or a receive timeout being exceeded by the remote host, or an underlying network resource issue. Local socket timeout was '10675199.02:48:05.4775807'.
```

That problem is fixed.

The problem you are reporting happens also in 19.3 and is certainly different.
Maybe the error message looks identical, but that is because many errors looks identical if the service tier crashes (then the socket connection is lost).
The problem is just not the loss of the socket connection - the problem happens before - and what I need to figure out is why the service tier crashes (hence - full script and full output is needed).

**Comment by JoelJame5** [comment]
@freddydk thanks, I can confirm this has been resolved in the latest build (20.0.36210.0).

**Comment by lvanvugt** [comment]
@JoelJame5, I can second that.
